packages/svelte/src/compiler/phases/3-transform/client/transform-client.js

```diff
@@ -457,7 +457,10 @@ export function client_component(analysis, options) {
 		analysis.uses_slots ||
 		analysis.slot_names.size > 0;
 
-	const body = [...module.body, ...state.hoisted];
+	// we hoist all the import declarations to the top of the file
+	const body = [...module.body, ...state.hoisted].sort((body_a, body_b) =>
+		body_a.type === 'ImportDeclaration' ? -1 : 1
```

Comment:
I wonder if you did something like this if it would stop it from re-ordering imports in the snapshots when not necessary? otherwise I'll note that `body_b` is currently unused and could be removed
```suggestion
		body_a.type === 'ImportDeclaration' && body_b.type !== 'ImportDeclaration' ? -1 : 1
```

Reply:
Uh I actually think this is better for multiple reasons: not only for the tests but also because someone might import in a certain order to have some side effect...will update code and tests as soon as I'm back at the desk.

Reply:
Yeah, ignoring `b`  seems to reverse the order of the lower part of the array:
```js
> [0,1,2,3,4,5,6,7,8,9].sort(a => a&1 ? -1 : 1)
> [ 1, 3, 5, 7, 9, 8, 6, 4, 2, 0 ] 
```

Reply:
Fixed it in the last commit